### PR TITLE
Fix issue related to environment and puppetclass field for Host Entity.

### PR DESCRIPTION
**Pull request by jameerpathan111**
##### Description of changes

Fixes issue related to `environment` and `puppetclass` field for `Host` and `SmartVariable` Entity.

##### Functional demonstration

```
(Pdb) entities.Host().create()
nailgun.entities.Host(all_parameters=[{'priority': 0, 'created_at': '2022-01-07 06:11:01 UTC', 'updated_at': '2022-01-07 06:11:01 UTC', 'id': 3, 'name': 'host_registration_remote_execution', 'parameter_type': 'boolean', 'value': True}, {'priority': 0, 'created_at': '2022-01-07 06:09:33 UTC', 'updated_at': '2022-01-07 06:09:33 UTC', 'id': 1, 'name': 'host_registration_insights', 'parameter_type': 'boolean', 'value': True}, {'priority': 0, 'created_at': '2022-01-07 06:11:01 UTC', 'updated_at': '2022-01-07 06:11:01 UTC', 'id': 4, 'name': 'host_packages', 'parameter_type': 'string', 'value': ''}, {'priority': 0, 'created_at': '2022-01-07 06:11:02 UTC', 'updated_at': '2022-01-07 06:11:02 UTC', 'id': 6, 'name': 'enable-puppet5', 'parameter_type': 'string', 'value': 'true'}, {'priority': 0, 'created_at': '2022-01-07 06:11:02 UTC', 'updated_at': '2022-01-07 06:11:02 UTC', 'id': 5, 'name': 'enable-epel', 'parameter_type': 'string', 'value': 'false'}, {'priority': 0, 'created_at': '2022-01-07 06:10:59 UTC', 'updated_at': '2022-01-07 06:10:59 UTC', 'id': 2, 'name': 'ansible_roles_check_mode', 'parameter_type': 'boolean', 'value': False}], architecture=nailgun.entities.Architecture(id=4), build=False, capabilities=['build'], comment=None, compute_profile=None, compute_resource=None, domain=nailgun.entities.Domain(id=3), enabled=True, hostgroup=None, host_parameters_attributes=[], ip=None, location=nailgun.entities.Location(id=7), mac='58:5a:c2:a6:de:cf', managed=True, medium=nailgun.entities.Media(id=16), model=None, name='pimfxeoxxg.uvatldk1zk', operatingsystem=nailgun.entities.OperatingSystem(id=3), organization=nailgun.entities.Organization(id=8), owner=nailgun.entities.User(id=4), provision_method='build', ptable=nailgun.entities.PartitionTable(id=212), puppet_ca_proxy=None, puppet_proxy=None, realm=None, subnet=None, uuid=None, id=4, image=None, interface=[nailgun.entities.Interface(host=nailgun.entities.Host(id=4), id=4)], build_status_label='Installed', owner_type='User')
(Pdb) entities.Host().search(query={'search': f'name="dhcp-3-4.example.com"'})
[nailgun.entities.Host(architecture=nailgun.entities.Architecture(id=1), build=False, build_status_label='Installed', capabilities=['build'], comment=None, compute_profile=None, compute_resource=None, domain=nailgun.entities.Domain(id=1), enabled=True, hostgroup=None, image=None, ip='1.2.3.4', location=nailgun.entities.Location(id=2), mac='6:f:d:2:c:7', managed=False, medium=None, model=nailgun.entities.Model(id=1), name='dhcp-3-4.example.com', operatingsystem=nailgun.entities.OperatingSystem(id=1), organization=nailgun.entities.Organization(id=1), owner=None, provision_method='build', ptable=None, puppet_ca_proxy=None, puppet_proxy=None, realm=None, subnet=None, uuid=None, id=2)]

```

### CODE REVIEW

**Comment by JacobCallahan** [comment]
While I think there is merit to this approach, perhaps it would be easier and more transparent to add a config setting for disabled fields.

**Review by swadeley — APPROVED** [review]
ACK, tested OK

File: nailgun/entities.py [hunk 1]
```diff
@@ -187,6 +188,13 @@ def _get_version(server_config):
     return getattr(server_config, 'version', Version('1!0'))
 
 
+@lru_cache()
```
Comment (JacobCallahan) [inline nailgun/entities.py:191]:
```suggestion
@lru_cache
```
non-blocking

**Review by JacobCallahan — APPROVED** [review]

**Review by jyejare — APPROVED** [review]

**Review by swadeley — COMMENTED** [review]
ACK

**Review by vsedmik — COMMENTED** [review]
ACK

**Pull request merged**